### PR TITLE
Support validating first party JWT for platforms without third party JWT

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for: https://github.com/istio/istio/issues/20178

To make the PR shorter and easier to review, this PR focues on adding the support of validating first party JWT. More PRs will follow up on this topic.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: manifests/global.yaml [hunk 1]
```diff
@@ -564,6 +564,11 @@ global:
   # we may change the default value of this option as "citadel".
   pilotCertProvider: "kubernetes"
 
+  # Configure the policy for validating JWT.
+  # Currently, two options are supported: "trustworthy-jwt" and "legacy-jwt".
+  jwtPolicy: "trustworthy-jwt"
```
Comment (howardjohn) [inline manifests/global.yaml]: We used to have a setting `useTrustworthyJwt`, do you think it makes sense to go back to using the same setting?

If not, should we call this first and third party jwt?

File: manifests/istio-control/istio-discovery/files/injection-template.yaml [hunk 2]
```diff
@@ -192,6 +192,8 @@ template: |
       {{ toYaml .Values.global.proxy.lifecycle | indent 4 }}
     {{- end }}
     env:
+    - name: JWT_POLICY
+      value: {{ .Values.global.jwtPolicy }}
```
Comment (howardjohn) [inline manifests/istio-control/istio-discovery/files/injection-template.yaml]: Line 391 is going to fail if your cluster doesn't have this enabled. If you want to test it just run `kind create cluster`, that will start a cluster without support. No pods will start

**Review by howardjohn — COMMENTED** [review]
This only solves a small portion of the problem. We still mount the trustworthy jwt, which will fail on clusters without support, and we still send XDS config referencing it.

I would suggest looking at the PR that removed support and working backwards as a starting point possibly?

File: manifests/global.yaml (hunk 1 above)
Comment (lei-tang) [reply manifests/global.yaml]: Changed to call it first and third party JWT.
Here jwtPolicy is used to express different JWT policies; the binary setting of useTrustworthyJwt makes it awkward to express different JWT policies while reducing the code readability.

File: manifests/istio-control/istio-discovery/files/injection-template.yaml (hunk 2 above)
Comment (lei-tang) [reply manifests/istio-control/istio-discovery/files/injection-template.yaml]: Added {{- if eq .Values.global.jwtPolicy "third-party-jwt" }} before line 391.

**Comment by lei-tang** [comment]
@howardjohn To support first party JWT, there will be multiple PRs. To make the PR shorter and easier to review, this PR focues on adding the support of validating first party JWT.

File: manifests/istio-control/istio-discovery/templates/deployment.yaml [hunk 3]
```diff
@@ -245,9 +247,11 @@ spec:
           - name: sds-uds-path
             mountPath: /var/run/sds
             readOnly: true
+          {{- if eq .Values.global.jwtPolicy "third-party-jwt" }}
```
Comment (howardjohn) [inline manifests/istio-control/istio-discovery/templates/deployment.yaml]: you need to remove the volume too, not just the volume mount

**Review by howardjohn — APPROVED** [review]
lgtm other than my one comment. can be a followup pr

File: manifests/istio-control/istio-discovery/templates/deployment.yaml (hunk 3 above)
Comment (lei-tang) [reply manifests/istio-control/istio-discovery/templates/deployment.yaml]: Added the conditional flag to the volume too. The mounting part will be addressed in follow up PRs.

File: manifests/istio-policy/templates/deployment.yaml [hunk 4]
```diff
@@ -218,9 +218,11 @@ spec:
         - name: sds-uds-path
           mountPath: /var/run/sds
           readOnly: true
+        {{- if eq .Values.global.jwtPolicy "third-party-jwt" }}
```
Comment (myidpt) [inline manifests/istio-policy/templates/deployment.yaml]: Please also set condition for:
https://github.com/istio/istio/blob/19cc5c9f7f0d903f159175fb259f3e5982f91f3b/manifests/istio-policy/templates/deployment.yaml#L50

File: manifests/istio-telemetry/mixer-telemetry/templates/deployment.yaml [hunk 5]
```diff
@@ -219,9 +219,11 @@ spec:
         - name: sds-uds-path
           mountPath: /var/run/sds
           readOnly: true
+        {{- if eq .Values.global.jwtPolicy "third-party-jwt" }}
```
Comment (myidpt) [inline manifests/istio-telemetry/mixer-telemetry/templates/deployment.yaml]: Please also set condition for the "projected volume"

**Review by myidpt — APPROVED** [review]

File: manifests/istio-policy/templates/deployment.yaml (hunk 4 above)
Comment (lei-tang) [reply manifests/istio-policy/templates/deployment.yaml]: Done.

File: manifests/istio-telemetry/mixer-telemetry/templates/deployment.yaml (hunk 5 above)
Comment (lei-tang) [reply manifests/istio-telemetry/mixer-telemetry/templates/deployment.yaml]: Done.

**Pull request merged**